R/tidy.R

```diff
@@ -114,6 +114,6 @@ tidy_data_frame <- function(object, digits = NULL) {
 warn_singularities <- function(object) {
   if (object$rank < object$k) {
     singularities <- object$k - object$rank
-    message(sprintf('%i coefficient%s not defined because of singularities\n', singularities, ifelse(singularities > 1, 's', '')))
+    message(sprintf('%i coefficient%s not defined because the design matrix is rank deficient\n', singularities, ifelse(singularities > 1, 's', '')))
```

Comment:
you usually don't need/want to sprintf inside message, you can just do `message(singularities, " cofficient", plural, " blah blah...")` - this will auto generate format strings for i18n should we ever do that.

Reply:
Thanks, old habits die hard.